Add TypeScript definitions

TylorS opened the pull request:
Adds TypeScript definitions for the public API.

This depends on `most` having TS definitions so I'd suggest waiting until that is actually the case :+1:

TylorS committed:
add TypeScript definitions

briancavalier commented on type-definitions/dom-event.d.ts:
```diff
@@ -0,0 +1,46 @@
+import {Stream} from "most";
+
+export function domEvent(event: string, node: EventTarget, capture: boolean): Stream<Event>
```
Is there a way in TS to represent that `capture` is optional, and defaults to `false` when not provided?  Perhaps `capture:boolean = false` works, or perhaps just `capture = false` (hopefully TS could infer that it's a boolean!)?

TylorS replied:
Ahhh yess. It would be `capture?: boolean`. I'll make the changes soon

briancavalier replied:
It looks like TS supports default param values: https://www.stevefenton.co.uk/2013/05/typescript-default-parameters/

I think that matches both the es6 source and the intent more closely than an optional param. Let's go with a default value.

TylorS replied:
Even better

TylorS replied:
I get the error `A parameter initializer is only allowed in a function or constructor implementation`
When trying to use a default value

briancavalier replied:
Ah darn. I guess default values aren't allowed here. Ok, let's go with your original plan of just making it optional.

briancavalier commented on package.json:
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.0",
   "description": "Streamlined DOM Events for most.js",
   "main": "dist/dom-event.js",
+  "typings": "type-definitions/dom-event.d.ts",
```
Since the lib gets built into a single file and only that file is published to npm, we probably need to either 1) also add `type-definitions/*.ts` to `files` below, or 2) keep the type definitions in `dist` alongside the built file, and change `files` to include `dist/*`.

Any thoughts on which of those seems better?

TylorS replied:
I don't have any fact or reasoning to back this, but I feel like 2 is the better option

TylorS committed:
chore(): most ts definitions to dist/

TylorS committed:
chore(package): adjust typings and add def file to `files`

Adds `dist/dom-event.d.ts` as a value to `files`
Points `typings` to the new location of TS def file

briancavalier commented:
Now that [most 0.19.6](https://github.com/cujojs/most/releases/tag/0.19.6) is released with TS defs, we can probably get back to this.  I'll try to take a look soon.

TylorS commented:
I think these should still be valid

briancavalier commented:
There are 2 copies of the type definition file in this PR right now (they differ slightly as well).  Do we need both?

TylorS commented:
Are there? There shouldnt be a need for 2, I'll fix it :)

TylorS commented:
I will have to close and reopen this, I deleted the fork this originated from.

TylorS commented:
Closing in favor of #8
